Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Michael Rothgang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Rothgang
+-/
+
+import Batteries.Data.String.Basic
+import Mathlib.Init.Data.Nat.Notation
+
+/-!
+## Text-based linters
+
+This file defines various mathlib linters which are based on reading the source code only.
+In practice, only style linters will have this form.
+All of these have been rewritten from the `lint-style.py` script.
+
+For now, this only contains the linter for too large files:
+further linters will be ported in subsequent PRs.
+
+-/
+
+open Lean Elab
+
+/-- Possible errors that text-based linters can report. -/
+-- We collect these in one inductive type to centralise error reporting.
+inductive StyleError where
+  /-- The current file was too large: this error contains the current number of lines
+  as well as a size limit (slightly larger). On future runs, this linter will allow this file
+  to grow up to this limit. -/
+  | fileTooLong (number_lines : ℕ) (new_size_limit : ℕ) : StyleError
+  deriving BEq
+
+/-- Create the underlying error message for a given `StyleError`. -/
+def StyleError.errorMessage (err : StyleError) : String := match err with
+  | StyleError.fileTooLong current_size size_limit =>
+      s!"{size_limit} file contains {current_size} lines, try to split it up"
```

Comment:
For reviewers: this matches the current output of `lint-style.py`. #13623 will add a more human-readable variant.